### PR TITLE
Feature allow glob pattern for patterns_dir

### PR DESCRIPTION
**Pull request by breml**
Added the possibility to specify a glob pattern for the pattern files in patterns_dir.

May be a solution for #33, without breaking compatibility.

### CODE REVIEW

**Comment by breml** [comment]
Could please someone give me some feedback?
I will fix the conflict, if there is some interest in merging this one.

**Comment by breml** [comment]
Resolved the conflict, please have a look.

File: lib/logstash/filters/grok.rb [hunk 1]
```diff
@@ -180,6 +180,10 @@ class LogStash::Filters::Grok < LogStash::Filters::Base
     #     NUMBER \d+
     config :patterns_dir, :validate => :array, :default => []
 
+    # Glob pattern, used to select the pattern files in the directories
+    # specified by patterns_dir
+    config :patterns_files_glob, :validate => :string, :default => "*"
```
Comment (jordansissel) [inline lib/logstash/filters/grok.rb]: It feels weird to have two settings to specify the files to load. While I see this new setting as needed for backwards compatibility, I wonder if we could add glob support to patterns_dir instead, and later renaming it to `patterns_path` -- thoughts on this approach?

(I'm open to deferring the patterns_path stuff to another time outside this PR)

**Comment by jordansissel** [comment]
jenkins, test it

File: spec/filters/grok_spec.rb [hunk 2]
```diff
@@ -710,6 +710,32 @@
...
   end
 
+  describe "patterns with file glob" do
+    require 'tmpdir'
+    require 'tempfile'
+
+    let(:tmpdir) { Dir.mktmpdir(nil, "/tmp") }
+
+    before do
+      @file3 = Tempfile.new(['grok', '.pattern'], tmpdir);     @file3.write('WORD \b[0-1]\b'); @file3.close
```
Comment (jordansissel) [inline spec/filters/grok_spec.rb]: Can you make this 3 lines? (replace ; with a new line)

File: spec/filters/grok_spec.rb [hunk 3]
```diff
@@ -710,6 +710,32 @@
...
+      @file4 = Tempfile.new(['grok', '.pattern.old'], tmpdir); @file4.write('WORD \b[2-5]\b'); @file4.close
```
Comment (jordansissel) [inline spec/filters/grok_spec.rb]: Same here, make this 3 lines (new file, write file, close file)

**Comment by jordansissel** [comment]
Code LGTM. Tests pass for me.

**Comment by jordansissel** [comment]
(I'll fix some of the style concerns post-merge)

**Comment by jordansissel** [comment]
Merged into master after rebasing and doing some style cleanup (faf47b3b66015d95d37cd10f9b4d72c69216714a)

Thank you :)

**Pull request closed without merging**